plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/SizeComputationExpressionTest.java

```diff
@@ -0,0 +1,286 @@
+import org.eclipse.sirius.tests.support.api.GraphicTestsSupportHelp;
+import org.eclipse.sirius.tests.swtbot.support.api.AbstractSiriusSwtBotGefTestCase;
+import org.eclipse.sirius.tests.swtbot.support.api.business.UILocalSession;
+import org.eclipse.sirius.tests.swtbot.support.api.business.UIResource;
+import org.eclipse.sirius.tests.swtbot.support.api.condition.CheckEditPartMoved;
+import org.eclipse.sirius.tests.swtbot.support.api.condition.CheckEditPartResized;
+import org.eclipse.sirius.tests.swtbot.support.api.condition.CheckSelectedCondition;
+import org.eclipse.sirius.tests.swtbot.support.api.condition.TextEditorAppearedCondition;
+import org.eclipse.sirius.tests.swtbot.support.api.editor.SWTBotSiriusDiagramEditor;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditPart;
+import org.junit.Assert;
+
+/**
+ * This class tests different configurations of the size computation expression.
+ * 
+ * @author gplouhinec
+ */
+public class SizeComputationExpressionTest extends AbstractSiriusSwtBotGefTestCase {
+
+    private UIResource sessionAirdResource;
+
+    private UILocalSession localSession;
+
+    private static final String DATA_UNIT_DIR = "data/unit/size/github-issue-65/"; //$NON-NLS-1$
+
+    private static final String MODEL = "github-issue-65.ecore"; //$NON-NLS-1$
+
+    private static final String SESSION_FILE = "github-issue-65.aird"; //$NON-NLS-1$
+
+    private static final String FILE_DIR = "/"; //$NON-NLS-1$
+
+    private static final String VSM_FILE = "github-issue-65.odesign"; //$NON-NLS-1$
+
+    private static final String REPRESENTATION_DESCRIPTION_NAME = "github-issue-65"; //$NON-NLS-1$
+
+    private static final String REPRESENTATION_NAME = "new " + REPRESENTATION_DESCRIPTION_NAME; //$NON-NLS-1$
+
+    private static final int DEFAULT_WIDTH_OR_HEIGHT = 0;
+
+    private static final int CUSTOM_WIDTH_OR_HEIGHT = 5;
+
+    private static final int AUTO_SIZE_MODE = -1;
+
+    private static final int CUSTOM_SIZE_COMPUTATION_EXPRESSION = 12;
+
+    private static final int WIDTH_AUTO_SIZED = 101;
+
+    private static final int HEIGHT_AUTO_SIZED = 51;
+
+    private static final int WIDTH_AFTER_RENAME = 263;
+
+    private static final String NEW_ECLASS_NAME = "EClass01234567890123456789"; //$NON-NLS-1$
+
+    @Override
+    protected void onSetUpBeforeClosingWelcomePage() throws Exception {
+        copyFileToTestProject(Activator.PLUGIN_ID, DATA_UNIT_DIR, MODEL, SESSION_FILE, VSM_FILE);
+    }
+
+    @Override
+    protected void onSetUpAfterOpeningDesignerPerspective() throws Exception {
+        sessionAirdResource = new UIResource(designerProject, FILE_DIR, SESSION_FILE);
+        localSession = designerPerspective.openSessionFromFile(sessionAirdResource);
+        editor = (SWTBotSiriusDiagramEditor) openRepresentation(localSession.getOpenedSession(), REPRESENTATION_DESCRIPTION_NAME, REPRESENTATION_NAME, DDiagram.class);
+    }
+
+    public void testAutoSizeComputationExpression() {
```

Comment:
Does this test check that it is correctly resized when the label is displayed on more than one line?
I did not see any test change since your last update that allow multiline label

Reply:
OK done. I changed the label `EClass01234567890123456789` with `EClass 01234567890123456789`, it's now on two lines:
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/33932970/c716e635-93dc-4244-b5c5-2e07728caf24)